### DISCUSSION TITLE
observation of  esp now and sbc decoder

### DISCUSSION THREAD

**Discussion by aqildad-create** [pschatzmann/arduino-audio-tools#794] [Q&A] [answered]
I am getting around 330/=sample per seconds so 330x250 bytes = 82,500 bytes per seconds with out your esp api ? Is this normal ?

I notice with sbc encoder at receiver when ever sample rate decrease from 330 to 300 samples then audio distorted . and sbc encoder at receiver need around 330 sample to run quality audio .. Is this normal ?

also sample rate changes from 330 sample to 300 sample to audio quality is changing accordingly . Is this normal ?

If the sample rate further decrease to 250 samples some time then audio decrease more .. so how to handle this . buffer is not a solution because if sample decrease from 300 sample even if their is buffer still audio is distorted . how to mange this ?

and why some time esp date rate decreased ? Is this normal ?

How to get high samples rate ?

**Comment by pschatzmann** [comment, marked as answer]
I don't remember the max transmission speed of esp-now. But that's the reason why I added classes that let you measure the thruput and google will be able to prove this answer. 

I don't understand your numbers and they to not make any sense to me. 

I don't see how you would get any reasonble audio below 8000 samples per second: so 330 is nonsense!
I also don't understand where the 250 bytes would come from. A sample usually has 2 bytes (at 16 bits_per_second) and you would multiply this by channels (which is 1 or 2).

**Comment by aqildad-create** [comment]
330 is espnow sample rate  each sample is of 250 bytes ..so total data to send is 330x250=82,500 and its 16 bit data so total samples of audio is 82500/2= 41,250/= samples of audio per seconds . 

so why  need to send 41,250/ bytes of data even with encoder (SBC)   . 

 I  was expecting half of data to send because SBC encoder must compressed the data .

**Comment by pschatzmann** [comment]
espnow does not have any sample rate and to call the max body size a sample is misleading

**Reply by aqildad-create** [reply]
yes i agree its not sample rate ,  82500 bytes is maximum espnow data rate i achieved but its almost equal to raw data processing .
